Unify copies of Version.xcconfig

ddkilzer opened the pull request:
#### 4b9a8ee9163f77e9ce6d1733c6f109a7de3034dd
<pre>
Unify copies of Version.xcconfig
<a href="https://bugs.webkit.org/show_bug.cgi?id=244450">https://bugs.webkit.org/show_bug.cgi?id=244450</a>
&lt;rdar://99249045&gt;

Reviewed by Alexey Proskuryakov.

- Move latest copy of Version.xcconfig into Configurations
  folder.
- Remove duplicate copies of Version.xcconfig in each project.
- Remove Version.xcconfig from Xcode projects.
- Change references to Version.xcconfig in other xcconfig files
  to use a relative path to the single copy.
- Make &quot;Update Info.plist with version information&quot; build script
  phases consistent between projects by using SRCROOT.

* Configurations/Version.xcconfig: Rename from Source/WebCore/PAL/Configurations/Version.xcconfig.
* Source/JavaScriptCore/Configurations/JSC.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/Configurations/TestAPI.xcconfig:
* Source/JavaScriptCore/Configurations/ToolExecutable.xcconfig:
* Source/JavaScriptCore/Configurations/Version.xcconfig: Remove.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/AngleMetalLib.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/Version.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/Version.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/Version.xcconfig: Remove.
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Version.xcconfig: Remove.
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/Version.xcconfig: Remove.
* Source/WebInspectorUI/Configurations/WebInspectorUIFramework.xcconfig:
* Source/WebInspectorUI/WebInspectorUI.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKit/Configurations/Version.xcconfig: Remove.
* Source/WebKit/SwiftOverlay/Configurations/Base.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/PlatformWin.cmake:
- Update script to use Configurations/Version.xcconfig.
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/Version.xcconfig: Remove.
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/Misc/WebKitVersionChecks.h:
- Update path to Version.xcconfig.
* Source/WebKitLegacy/scripts/generate-webkitversion.pl: Ditto.
(printLicenseHeader):
* Source/cmake/OptionsMac.cmake: Ditto.
- Add comment about PlatformWin.cmake generating WebKitVersion.h
  since it parses Version.xcconfig.
- Update local WebKit version to match Version.xcconfig.

Canonical link: <a href="https://commits.webkit.org/253909@main">https://commits.webkit.org/253909@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2985b073d2cd86b9b7906c5dd30baab7ad1ef257

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87362 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31451 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18173 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96591 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149937 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91337 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29814 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26013 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79479 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91364 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92979 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24089 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74154 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23927 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79059 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79284 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66943 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/79214 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27530 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13118 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/72895 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27482 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14133 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/25961 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2747 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29169 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/36993 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/75715 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29100 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33410 "Passed tests") | [❌ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/16775 "Found 2 new JSC stress test failures: stress/call-apply-exponential-bytecode-size.js.mini-mode, stress/call-apply-exponential-bytecode-size.js.no-llint") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Looks fine as far as I'm concerned, but please get a sign-off from Alan, to confirm the submissions don't break.

Commit:
Unify copies of Version.xcconfig
https://bugs.webkit.org/show_bug.cgi?id=244450
<rdar://99249045>

Reviewed by Alexey Proskuryakov.

- Move latest copy of Version.xcconfig into Configurations
  folder.
- Remove duplicate copies of Version.xcconfig in each project.
- Remove Version.xcconfig from Xcode projects.
- Change references to Version.xcconfig in other xcconfig files
  to use a relative path to the single copy.
- Make "Update Info.plist with version information" build script
  phases consistent between projects by using SRCROOT.

* Configurations/Version.xcconfig: Rename from Source/WebCore/PAL/Configurations/Version.xcconfig.
* Source/JavaScriptCore/Configurations/JSC.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/Configurations/TestAPI.xcconfig:
* Source/JavaScriptCore/Configurations/ToolExecutable.xcconfig:
* Source/JavaScriptCore/Configurations/Version.xcconfig: Remove.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/AngleMetalLib.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/Version.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/Version.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/Version.xcconfig: Remove.
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Version.xcconfig: Remove.
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/Version.xcconfig: Remove.
* Source/WebInspectorUI/Configurations/WebInspectorUIFramework.xcconfig:
* Source/WebInspectorUI/WebInspectorUI.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKit/Configurations/Version.xcconfig: Remove.
* Source/WebKit/SwiftOverlay/Configurations/Base.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/PlatformWin.cmake:
- Update script to use Configurations/Version.xcconfig.
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/Version.xcconfig: Remove.
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/Misc/WebKitVersionChecks.h:
- Update path to Version.xcconfig.
* Source/WebKitLegacy/scripts/generate-webkitversion.pl: Ditto.
(printLicenseHeader):
* Source/cmake/OptionsMac.cmake: Ditto.
- Add comment about PlatformWin.cmake generating WebKitVersion.h
  since it parses Version.xcconfig.
- Update local WebKit version to match Version.xcconfig.

Canonical link: https://commits.webkit.org/253909@main
